### PR TITLE
Added user event callbacks

### PR DESCRIPTION
**Pull request by frederickk**
I made some tweaks to this for my [Stjörnuíþrótt](https://github.com/frederickk/stjoernuithrott) drone to allow for user event handling in addition to how passthrough handles midi. I thought it might be of interest to others.

### CODE REVIEW

File: example-event.lua [hunk 1]
```diff
@@ -0,0 +1,23 @@
+local Passthrough = include("lib/passthrough")
+
+
+function midi_device_event(data)
+  local msg = midi.to_msg(data)
+  print("device")
+  tab.print(msg)
+end
+
+function midi_output_event(data)
```
Comment (nattog) [inline example-event.lua]:
```suggestion
function midi_interface_event(data)
```

File: example-event.lua [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  local msg = midi.to_msg(data)
+  print("out")
+  tab.print(msg)
+end
+
+function init()
+    Passthrough.init()
+    Passthrough.user_device_event = midi_device_event
+    Passthrough.user_output_event = midi_output_event
```
Comment (nattog) [inline example-event.lua]:
```suggestion
    Passthrough.user_interface_event = midi_interface_event
```

**Review by nattog — CHANGES_REQUESTED** [review]
Thanks for this, certainly a nice feature to add! I'll try it out on the hardware tomorrow, just one small thing at a first glance.

File: example-event.lua (hunk 2 above)
Comment (frederickk) [reply example-event.lua]: Good catch! Left that in there from my debugging.

**Comment by frederickk** [comment]
I amended with PR with the changes requested.

**Review by nattog — APPROVED** [review]
Works very nicely, thank you!

**Pull request merged**